### DISCUSSION TITLE
How to align vertically the content of a div?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#11437] [General - Components / Directives / etc] [answered]
I have a `row` of QButtons and wanted to add a `div` with some text:

```
        <q-btn :text-color='note.pinned?"blue":"grey"' class="vertical-bottom" flat icon='bi-pin-angle-fill' @click='note.pinned=!note.pinned'>
          <q-tooltip>pin note to top</q-tooltip>
        </q-btn>
        <q-btn :color='note.archived?"blue":"grey"' flat icon="archive" @click='note.archived=!note.archived'></q-btn>
        <q-btn flat icon="delete">
(...)
        </q-btn>
        <div v-if="note.deleteOn">deleted {{ DateTime.fromISO(note.deleteOn).toRelativeCalendar() }}</div>
```

The result is:

![image](https://user-images.githubusercontent.com/1345886/142770329-361fa0e6-9ed4-47d4-92c8-b56faea62879.png)

The `div`  took the whole vertical space (which is fine) but the text inside is justified to top.

I tried so many things (all the Flexbox justify classes for elements and text (many examples [here](https://stackoverflow.com/questions/25311541/how-to-vertically-align-text-inside-a-flexbox), the Quasar provided classes and whatnot) and simply could not make this text center vertically (cross-axis relative to the `row`). How to do that?

**Comment by hawkeye64** [comment, marked as answer]
Use `items-center`

**Reply by wsw70** [reply]
Unfortunately

    <div v-if="note.deleteOn" class="items-center">deleted {{DateTime.fromISO(note.deleteOn).toRelativeCalendar()}}</div>

does not help. The text is still justified to the top.

![image](https://user-images.githubusercontent.com/1345886/142776958-a48469ae-fa2e-41f6-98c4-e5b286fb9700.png)

**Reply by hawkeye64** [reply]
You need `row justify-center items-center` on the parent div, not the child. I think for child it's something like `self-center` but you would have to verify.

**Reply by wsw70** [reply]
@hawkeye64 ah fantastic - `self-centered` did the trick. Thank you very much!
